### PR TITLE
Update gradient accumulation example docs for plugin and fix bf16 test for mps

### PR DESCRIPTION
**Pull request by iantbutler01**
Solves issue #1460 by adding more documentation.

In addition while running testing, a mixed precision bf16 test failed. Normally I work with accelerate on a linux box, but I was doing the doc edits and caught that it was trying to do mixed precision bf16 on a M1 laptop which defaults to the mps device type.

I added a condition in accelerator.py to fail for mps devices as bf16 doesn't seem to be supported on them anyway, see the below pytorch issue for more information
https://github.com/pytorch/pytorch/issues/78168

I can understand if you'd rather me split that test fix one out into another PR or that you'd prefer a different approach all together.

### CODE REVIEW

**Comment by sgugger** [comment]
If bfloat16 is not supported on M1, we should probably error sooner when the Accelerator is created and combine bfloat16 with mps. Then we can also skip the test for bfloat16 if on an M1. So all in all it's probably better to make a new PR for all of this and focus this one the docs :-)

**Comment by iantbutler01** [comment]
Sounds good, I'll break this apart, and rework the fix and open another PR for it!

**Comment by iantbutler01** [comment]
Okay, now this is just the docs @sgugger I will circle back around in the next few days with the other PR.

**Review by sgugger — APPROVED** [review]
Thanks!

**Review by muellerzr — APPROVED** [review]
Thanks for this!
